frontend/src/data/variables/AcsHealthInsuranceProvider.ts

```diff
@@ -0,0 +1,120 @@
+  constructor() {
+    super("acs_health_insurance_provider", [
+      "health_insurance_count",
+      "health_insurance_per_100k",
+    ]);
+  }
+
+  getDatasetId(breakdowns: Breakdowns): string {
+    if (breakdowns.hasOnlySex()) {
+      return breakdowns.geography === "county"
+        ? "acs_health_insurance-health_insurance_by_sex_county"
+        : "acs_health_insurance-health_insurance_by_sex_state";
+    }
+
+    if (breakdowns.hasOnlyRace()) {
+      return breakdowns.geography === "county"
+        ? "acs_health_insurance-health_insurance_by_race_county"
+        : "acs_health_insurance-health_insurance_by_race_state";
+    }
+
+    // Age only breakdown is not supported yet, due to the dataset not being
+    // Aggregated on the backend.
+    throw new Error("Not implemented");
+  }
+
+  async getDataInternal(
+    metricQuery: MetricQuery
+  ): Promise<MetricQueryResponse> {
+    const breakdowns = metricQuery.breakdowns;
+    const datasetId = this.getDatasetId(breakdowns);
+    const acsDataset = await getDataManager().loadDataset(datasetId);
+
+    let df = acsDataset.toDataFrame();
+
+    // If requested, filter geography by state or county level
+    // We apply the geo filter right away to reduce subsequent calculation times
+    df = this.filterByGeo(df, breakdowns);
+    df = this.renameGeoColumns(df, breakdowns);
+    df = df.renameSeries({ race: "race_and_ethnicity" });
+    df = df.parseInts([
+      "with_health_insurance",
+      "without_health_insurance",
+      "total_health_insurance",
+    ]);
+
+    if (breakdowns.geography === "national") {
+      df = df
+        .pivot([breakdowns.getSoleDemographicBreakdown().columnName], {
+          fips: (series) => USA_FIPS,
+          fips_name: (series) => USA_DISPLAY_NAME,
+          with_health_insurance: (series) => series.sum(),
+          without_health_insurance: (series) => series.sum(),
+          total_health_insurance: (series) => series.sum(),
+        })
+        .resetIndex();
+    } else {
+      df = df.pivot(
+        [
+          "fips",
+          "fips_name",
+          breakdowns.getSoleDemographicBreakdown().columnName,
+        ],
+        {
+          with_health_insurance: (series) => series.sum(),
+          without_health_insurance: (series) => series.sum(),
+          total_health_insurance: (series) => series.sum(),
+        }
+      );
+    }
+
+    let totalPivot: { [key: string]: (series: ISeries) => any } = {
+      with_health_insurance: (series: ISeries) => series.sum(),
+      without_health_insurance: (series: ISeries) => series.sum(),
+      total_health_insurance: (series: ISeries) => series.sum(),
+    };
+
+    totalPivot[breakdowns.getSoleDemographicBreakdown().columnName] = (
+      series: ISeries
+    ) => TOTAL;
+
+    // Calculate totals where dataset doesn't provide it
+    // TODO- this should be removed when Totals come from the Data Server
+    const total = df.pivot(["fips", "fips_name"], totalPivot).resetIndex();
```

Comment:
I would just put the value directly here rather than pulling it out into totalPivot variable